LICENSE

```diff
@@ -1,339 +1,72 @@
-    GNU General Public License for more details.
-
-    You should have received a copy of the GNU General Public License along
-    with this program; if not, write to the Free Software Foundation, Inc.,
-    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
-
-Also add information on how to contact you by electronic and paper mail.
-
-If the program is interactive, make it output a short notice like this
-when it starts in an interactive mode:
-
-    Gnomovision version 69, Copyright (C) year name of author
-    Gnomovision comes with ABSOLUTELY NO WARRANTY; for details type `show w'.
-    This is free software, and you are welcome to redistribute it
-    under certain conditions; type `show c' for details.
-
-The hypothetical commands `show w' and `show c' should show the appropriate
-parts of the General Public License.  Of course, the commands you use may
-be called something other than `show w' and `show c'; they could even be
-mouse-clicks or menu items--whatever suits your program.
-
-You should also get your employer (if you work as a programmer) or your
-school, if any, to sign a "copyright disclaimer" for the program, if
-necessary.  Here is a sample; alter the names:
-
-  Yoyodyne, Inc., hereby disclaims all copyright interest in the program
-  `Gnomovision' (which makes passes at compilers) written by James Hacker.
-
-  <signature of Ty Coon>, 1 April 1989
-  Ty Coon, President of Vice
-
-This General Public License does not permit incorporating your program into
-proprietary programs.  If your program is a subroutine library, you may
-consider it more useful to permit linking proprietary applications with the
-library.  If this is what you want to do, use the GNU Lesser General
-Public License instead of this License.
+FunGame License
+Version 0.1.0
+
+Preamble
+
+A video game is a multi-media artwork. The software and the game art must necessarily be
+licensed under different license agreements. There are five general parts 1) the software
+2) the game art 3) the game play 4) promotional material, and 5) additional artwork. Each
+is licensed under a different license. (At this point only the first three exist). 
+
+In the spirit of GNU the software is free as in bear and free as in speech. You are free
+to use the artwork for your own personal use. You may not however modify and reproduce
+the artwork.
+
+This license is not fixed. As it stands this license prevents the use of "texture packs."
+For this reason, the repository is also licensed under any potential future versions of
+this license.
+
+The purpose of this license is to describe the license for specific material in this
+repository. If not specified it's probably GNU. NOTE: the three license files do not
+necessarily describe all the material in this repository. For example, the vendor
+subdirectory contains external material that has its own license. I addition there is
+code copied for Stack Overflow. (At the time of writing only two copied sections)
```

Comment:
I believe all SO code is implicitly licensed under some CC license compatible with GPLv2, but double check.

Reply:
The only thing copied directly from stack overflow is something you added. 

It's mostly compatible. There are some slight differences

Reply:
here found it 

https://stackoverflow.com/help/licensing

https://www.gnu.org/licenses/license-list.en.html#ccbysa